panml/utils.py

```diff
@@ -0,0 +1,89 @@
+# Dependencies for data processing and general machine learning implementations
```

Comment:
I recommend calling it more specific than utils. Maybe `vector-engine` or something. Utils generally become a dumping ground for random code. Best to be avoided

Reply:
ok ill fix